opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.hpp

```diff
@@ -36,63 +36,65 @@ namespace Opm {
 
     class SummaryNode {
     public:
-        SummaryNode(const std::string& keyword) :
-            ecl_node(0, keyword.c_str(), "UNIT", 0.0)
-        {}
-
-
-        SummaryNode(const std::string& keyword, const std::string& wgname) :
-            ecl_node(0, keyword.c_str(), wgname.c_str(), "UNIT", 0.0, ":")
-        {}
-
-        SummaryNode(const std::string& keyword, const std::string& wgname, int num) :
-            ecl_node(0, keyword.c_str(), wgname.c_str(), num, "UNIT", 0.0, ":")
-        {}
-
-        SummaryNode(const std::string& keyword, int num) :
-            ecl_node(0, keyword.c_str(), num, "UNIT", 0, ":")
-        {}
-
-        SummaryNode(const std::string& keyword, int num, const int grid_dims[3]) :
-            ecl_node(0, keyword.c_str(), num, "UNIT", grid_dims, 0, ":")
-        {}
+        enum class Category {
+            Well, Group, Field,
+            Region, Block,
+            Connection, Segment,
+            Miscellaneous,
+        };
+
+        enum class Type {
+            Rate, Total, Ratio, Pressure, Count,
+            Undefined,
+        };
+
+        explicit SummaryNode(std::string keyword, const Category cat);
+
+        SummaryNode& parameterType(const Type type);
+        SummaryNode& namedEntity(std::string name);
+        SummaryNode& numericID(const int num);
+        SummaryNode& isUserDefined(const bool userDefined);
+
+        const std::string& keyword() const { return this->keyword_; }
+        Category category() const { return this->category_; }
+        Type type() const { return this->type_; }
+        const std::string& namedEntity() const { return this->name_; }
+        int numericID() const { return this->numericID_; }
+        bool isUserDefined() const { return this->userDefined_; }
+
+        std::string uniqueNodeKey() const;
 
-        SummaryNode(const std::string& keyword, const std::string& wgname, int num, const int grid_dims[3]) :
-            ecl_node(0, keyword.c_str(), wgname.c_str(), num, "UNIT", grid_dims, 0, ":")
-        {}
-
-        std::string wgname() const {
-            const char * c_ptr = this->ecl_node.get_wgname();
-            if (c_ptr)
-                return std::string(c_ptr);
-            else
-                return "";
-        }
-
-        std::string keyword() const {
-            return this->ecl_node.get_keyword();
-        }
-
-        std::string gen_key() const {
-            return this->ecl_node.get_gen_key1();
-        }
+    private:
+        std::string keyword_;
+        Category    category_;
+        Type        type_{ Type::Undefined };
+        std::string name_{};
+        int         numericID_{std::numeric_limits<int>::min()};
```

Comment:
This is in my opinon not a `numericID` - but rather a `numericValue`; it can not be used to look anything.

Reply:
> This is in my opinon not a `numericID` - but rather a `numericValue`; it can not be used to look anything.

I did mean ID as in "entity", not "index".  This will go into the `NUMS` array of the `.SMSPEC` file.  How about `numericEntity` instead?

Reply:
We have all got our pet pevees; names is one of mine. As I see it this is *just a number* - it is not an entity - i.e. a meaningful object to talk about in it's one right. Calling it `numericEntity` would be similar to call the `name` property `stringEntity`.

I'll shut up now!

Reply:
> As I see it this is _just a number_ - it is not an entity - i.e. a meaningful object to talk about in it's one right.

Okay, then I'm missing something.  I was under the impression that for the cases that has a non-trivial value (i.e., anything not in the well, group, field or "miscellaneous" categories), then this value refers to the actual object (e.g., the cell, the connection/completion, the region, the segment).  It does clearly become a little more involved in the case of a region-to-region parameter in which case it refers to a distinct pair of regions.  On the other hand, we do not currently support those parameters.

Reply:
OK - we are splitting hairs here. I agree with this:

> I was under the impression that for the cases that has a non-trivial value (i.e., anything not in the well, group, field or "miscellaneous" categories), then this value refers to the actual object (e.g., the cell, the connection/completion, the region, the segment

Reply:
> OK - we are splitting hairs here. I agree with this:
> 
> > I was under the impression that for the cases that has a non-trivial value (i.e., anything not in the well, group, field or "miscellaneous" categories), then this value refers to the actual object (e.g., the cell, the connection/completion, the region, the segment

I'll rename both the data member and the member functions to `number`.